apps/mobile-mzima-client/src/app/profile/information/information.page.ts

```diff
@@ -179,9 +208,197 @@ export class InformationPage {
+                    message: 'Profile photo updated successfully',
+                    duration: 3000,
+                    position: 'bottom',
+                  });
+                },
+                error: async (error) => {
+                  console.error('Failed to update profile photo. Please try again', error);
+                  this.isUploadInProgress = false;
+                  this.profilePhotoComponent.uploadingSpinner = false;
+                  await this.databaseService.remove(STORAGE_KEYS.PROFILE_PHOTO);
+                  this.toastService.presentToast({
+                    message: 'Failed to add profile photo',
+                    duration: 3000,
+                    position: 'bottom',
+                  });
+                },
+              });
+            } else {
+              const payload: any = {
+                config_key: configKey,
+                config_value: configValue,
+              };
+              this.usersService.postUserSettings(userId, payload).subscribe(
+                async () => {
+                  this.profilePhotoComponent.uploadingSpinner = false;
+                  this.profilePhotoComponent.hasUploadedPhoto = true;
+                  this.userPhoto = photoUrl;
+                  this.isUploadInProgress = false;
+                  this.toastService.presentToast({
+                    message: 'Profile photo updated successfully',
+                    duration: 3000,
+                    position: 'bottom',
+                  });
+                  await this.databaseService.remove(STORAGE_KEYS.PROFILE_PHOTO);
+                },
+                async (error) => {
+                  console.error('Failed to add profile photo', error);
+                  this.isUploadInProgress = false;
+                  this.profilePhotoComponent.uploadingSpinner = false;
+                  this.toastService.presentToast({
+                    message: 'Failed to add profile photo. Please try again',
+                    duration: 3000,
+                    position: 'bottom',
+                  });
+                  await this.databaseService.remove(STORAGE_KEYS.PROFILE_PHOTO);
+                },
+              );
+            }
+          });
+        } else {
+          console.error('User data or user ID is missing');
+          this.isUploadInProgress = false;
+          this.profilePhotoComponent.uploadingSpinner = false;
+        }
+      });
+  }
+
   public async updateUserInformation(): Promise<void> {
-    await this.updateFormAsync(this.form, { realname: this.form.value.realname! });
-    this.form.controls['role'].disable();
+    this.form.disable();
+    console.log('Starting update process');
+
+    try {
+      let photoUpdated = false;
+      let formUpdated = false;
+
+      if (this.isPhotoChanged) {
+        if (this.selectedFileKey && this.selectedCaption) {
+          await this.uploadPhoto(this.selectedFileKey, this.selectedCaption);
```

Comment:
I don't think you need the if-statement on row 371 since the filekey always seems to be the same (if I haven't missed anything). For the caption, is that used anywhere? If not, you could just use an empty string or something, if the API requires it.

Reply:
@Angamanga Alright Ive removed the unnecessary emission for the caption, the default 'image_upload' caption will now be used instead of emitting the name as the caption